crates/subspace-runtime/src/lib.rs

```diff
@@ -1048,7 +1048,9 @@ impl FraudProofStorageKeyProvider<NumberFor<Block>> for StorageKeyProvider {
             FraudProofStorageKeyRequest::DomainAllowlistUpdates(domain_id) => {
                 Messenger::domain_allow_list_update_storage_key(domain_id)
             }
-            FraudProofStorageKeyRequest::BlockDigest => sp_domains::system_digest_final_key(),
+            FraudProofStorageKeyRequest::DomainRuntimeUpgrades => {
+                pallet_domains::DomainRuntimeUpgrades::<Runtime>::hashed_key().to_vec()
```

Comment:
Should this be using `DomainRuntimeUpgradeRecords` or `DomainRuntimeUpgrades` for the parent block instead?

(`pallet_domains::Hooks::on_initialize()` clears the contents of `DomainRuntimeUpgrades` using `take()`.)

Reply:
The test is failing with `Invalid Domain extrinsic root proof verification failed: StorageProof(DomainRuntimeUpgradesStorageProof(MissingValue))` this is because  we clear the `DomainRuntimeUpgrades`

Reply:
How did you find this failure in the logs?

All I can see is a test hang (not a test failure):
https://github.com/autonomys/subspace/actions/runs/12459186519/job/34775706616?pr=3329#step:11:1535

Reply:
I run the test locally `cargo test -p domain-client-operator -- --exact tests::test_invalid_domain_extrinsics_root_proof_creation`

Reply:
> To fix it we need to instead reset the `DomainRuntimeUpgrades` to an empty vec so we can generate a proof-of-empty-value.

But an empty value doesn't work with the rest of the code, which needs a list of upgraded runtime IDs:
https://github.com/autonomys/subspace/pull/3329/files#diff-4828b750acc5f6c044ce5281811e58a5135ab6b80541bd805a1dde9281159f9fR344-R346

So I think we should be using `DomainRuntimeUpgradeRecords` instead.